File: Leetcode/66 - Plus One/solution.py
def plusOne(self, digits: List[int]) -> List[int]:
    i: int = len(digits) - 1
    carry: int = 1

    while i >= 0 and carry:
        digits[i] += carry
        carry = digits[i] // 10
        digits[i] %= 10
        i -= 1

    if carry:
        digits.insert(0, 1)

    return digits

File: Leetcode/66 - Plus One/solution.cpp
vector<int> plusOne(vector<int> &digits)
{
    int i = digits.size() - 1, carry = 1;
    do
    {
        digits[i] += carry;
        carry = digits[i] / 10;
        digits[i] %= 10;
        i--;
    } while (i >= 0 && carry);
    if (carry)
    {
        digits.insert(digits.begin(), 1);
    }
    return digits;
}